spring-integration-core/src/main/java/org/springframework/integration/channel/AbstractMessageChannel.java

```diff
@@ -407,7 +413,11 @@ public final boolean send(Message<?> message, long timeout) {
 			if (this.datatypes.length > 0) {
 				message = this.convertPayloadIfNecessary(message);
 			}
-			if (interceptors.getInterceptors().size() > 0) {
+			if (logger.isDebugEnabled()) {
+				logger.debug("preSend on channel '" + this + "', message: " + message);
+			}
+			boolean hasInterceptors = interceptors.getInterceptors().size() > 0;
```

Comment:
Right, I understand your solution, but there is no the "second invocation of getInterceptors().size()" ...